storagemarket/types.go

```diff
@@ -539,6 +540,17 @@ type DataRef struct {
 	PieceSize abi.UnpaddedPieceSize // Optional for non-manual transfer, will be recomputed from the data if not given
 }
 
+// ProviderDealState represents a Provider's current state of a deal
+type ProviderDealState struct {
+	State       StorageDealStatus
```

Comment:
Is there a reason this is not just a MinerDeal?

Reply:
It was originally to deal with the case where the miner deal was not found.  However, now that we have signature verification, we have to look up the miner deal to find the StorageClient's address.  So this struct should flattened.